### ISSUE TITLE
option --disable-pending-cops is activated with as little as --disable-p

### ISSUE THREAD

**Issue by matthewhively** [rubocop/rubocop#12280]
## Expected behavior

`rubocop --help` lists command: `--disable-pending-cops`
And invoking rubocop with that option does what it says it does.

## Actual behavior

`rubocop --disable-p` will also disable the pending cops.
`rubocop --disable-` or less will yield `ambiguous option: --disable-`

It looks like rubocop is using substring matches for all command line options?
Its kinda convenient to only type `--disable-pending` instead of the full option.
But this seems weird, I would assume any unrecognized options should throw an error, but I guess maybe this is intended and not a bug?

## RuboCop version

```
rubocop -V
1.57.1 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-darwin21]
  - rubocop-performance 1.19.1
  - rubocop-rails 2.21.2
```

**Comment by matthewhively** [comment]
Still occurs in rubocop 1.75.2

**Comment by matthewhively** [comment]
Still occurring in rubocop 1.79.1

**Comment by matthewhively** [comment]
Still occurring in rubocop 1.81.7

**Comment by dvandersluis** [comment]
This is a quirk of ruby's `OptionParser`, which is what rubocop uses to parse command line arguments. By default, `require_exact` mode is `false`, which allows the behaviour you're experiencing, partial, unambiguous switches are accepted.

We could set `require_exact` to `true`, but I don't see what the benefit would be. 

I opened #14680 to have rubocop handle ambiguous options the same way as invalid options, rather than spitting out a backtrace.
